### PR TITLE
Capture column level dependency for Set Operators

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

**Review by kokosing — APPROVED** [review]

**Review by ksobolew — APPROVED** [review]
From the unit tests I can understand what's the purpose of this change, but some kind of short description would still be useful

**Comment by kokosing** [comment]
Please wait for @kasiafi review before mergin.

File: core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java [hunk 1]
```diff
@@ -2219,6 +2219,14 @@ protected Scope visitSetOperation(SetOperation node, Optional<Scope> scope)
                         oldField.getOriginTable(),
                         oldField.getOriginColumnName(),
                         oldField.isAliased());
+
+                int index = i; // Variable used in Lambda should be final
+                analysis.addSourceColumns(
+                        outputDescriptorFields[index],
+                        childrenTypes.stream()
```
Comment (kasiafi) [inline core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java:2226]: For `INTERSECT`, and particularly for `EXCEPT`, we could report only the source columns from the leftmost relation.
I'm not sure of the intended semantics, though.

Comment (kokosing) [reply core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java:2226]: Well, it does not matter much. We care about columns being used in the operation. 

Also for security reasons if one column is restricted, and you know other columns then you could "calculate" the restricted column data. Notice the case `A EXCEPT B` where B is restricted and A contains the whole domain. Given that I would prefer that anything that is created from ``A EXCEPT B` would be also restricted.

Notice one could say that `coalesce(a, b, c)` is hitting the same issue when `a` is not nullable, and we might have more corner cases like this.

**Review by kasiafi — APPROVED** [review]

**Pull request merged**